src/Sylius/Behat/Context/Api/Shop/ProductContext.php

```diff
@@ -80,4 +116,23 @@ public function itsCurrentVariantShouldBeNamed(string $variantName): void
             )
         );
     }
+
+    private function hasProductWithPrice(array $resource, int $price, ?string $productCode = null): bool
+    {
+        foreach ($resource as $product) {
+            if ($productCode && $product['code'] !== $productCode) {
+                continue;
+            }
+
+            foreach ($product['variants'] as $variant) {
+                foreach ($variant['channelPricings'] as $channelPricing) {
+                    if ($channelPricing['price'] === $price) {
+                        return true;
+                    }
+                }
```

Comment:
```suggestion
                return $variant['channelPricings'][$channelCode]['price'] === $price;
```

Reply:
The current resolution will tell us only if there is at least one given price for the product(among all variants and channels)